codegens/objective-c/test/unit/convert.test.js

```diff
@@ -0,0 +1,130 @@
+var convert = require('../../index').convert,
+  expect = require('chai').expect,
+  collection = require('./fixtures/collection.json'),
+  sdk = require('postman-collection'),
+  expectedSnippets = require('./fixtures/snippets.json');
+
+describe('Objective-C-NSURLSession Converter', function () {
+  describe('convert for different request types', function () {
+    collection.item.forEach((item) => {
+      it(item.name, function (done) {
+        const request = new sdk.Request(item.request);
+        convert(request, {}, (err, snippet) => {
+          if (err) {
+            expect.fail(null, null, err);
+            return done();
+          }
+          expect(snippet).to.equal(expectedSnippets[item.name]);
+          return done();
+        });
+      });
+    });
+  });
+});
+
+describe('Options Tests', function () {
+  it('should indent snippet with type and count specified', function () {
+    var request = new sdk.Request(collection.item[0].request);
+    convert(request, {
+      indentType: 'Tab',
+      indentCount: 2
+    }, function (error, snippet) {
+      if (error) {
+        expect.fail(null, null, error);
+      }
+      expect(snippet).to.be.a('string');
+      var snippetArray = snippet.split('\n'),
+        i;
+      for (i = 0; i < snippetArray.length; i++) {
+        if (snippetArray[i].startsWith('cachePolicy:NSURLRequestUseProtocolCachePolicy')) {
+          expect(snippetArray[i + 1].charAt(0)).to.equal('\t');
+          expect(snippetArray[i + 1].charAt(1)).to.equal('\t');
+          expect(snippetArray[i + 1].charAt(2)).to.not.equal(' ');
+        }
+      }
+    });
+  });
+
+  it('should use all the default options', function () {
+    var request = new sdk.Request({
+      'method': 'POST',
+      'header': [],
+      'body': {
+        'mode': 'raw',
+        'raw': '  trim this body  '
+      },
+      'url': {
+        'raw': 'https://postman-echo.com/post',
+        'protocol': 'https',
+        'host': [
+          'postman-echo',
+          'com'
+        ],
+        'path': [
+          'post'
+        ]
+      }
+    });
+    convert(request, {}, function (error, snippet) {
+      if (error) {
+        expect.fail(null, null, error);
+      }
+      expect(snippet).to.be.a('string');
+      var snippetArray = snippet.split('\n'),
+        i;
+      for (i = 0; i < snippetArray.length; i++) {
+        if (snippetArray[i].startsWith('cachePolicy:NSURLRequestUseProtocolCachePolicy')) {
+          expect(snippetArray[i + 1].charAt(0)).to.equal(' ');
+          expect(snippetArray[i + 1].charAt(1)).to.equal(' ');
+          expect(snippet).to.include('timeoutInterval:10');
```

Comment:
Is 10 used as infinity?

Reply:
I couldn't find a way to set infinite request time out for nsurlsession. So setting a 10 sec convention I found in many sample snippets used.

Reply:
Ah, okay!